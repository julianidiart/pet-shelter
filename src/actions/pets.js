import uuid from "uuid";
import database from "../firebase";

// ADD_PET
export const addPet = pet => ({
  type: "ADD_PET",
  pet
});

export const startAddPet = (petData = {}) => {
  return dispatch => {
    const { name = "" } = petData;
    const pet = { name };
    database
      .ref("pet")
      .push(pet)
      .then(ref => {
        dispatch(addPet({ id: ref.key, ...pet }));
      });
  };
};

// REMOVE_PET
export const removePet = id => ({
  type: "REMOVE_PET",
  id
});

// EDIT_PET
export const editPet = (id, pet) => ({
  type: "EDIT_PET",
  id,
  pet
});
