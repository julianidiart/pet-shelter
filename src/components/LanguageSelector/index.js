import React, { Component } from "react";
import LanguageContext from "../../contexts/LanguageContext";
import unitedStatesFlag from "../../assets/images/united-states.png";
import italyFlag from "../../assets/images/italy.png";
import spainFlag from "../../assets/images/spain.png";

class LanguageSelector extends Component {
  static contextType = LanguageContext;

  render() {
    return (
      <div className="header__content header__content--flags">
        <img
          src={unitedStatesFlag}
          alt="english"
          onClick={() => this.context.onLanguageChange("en")}
        />
        <img
          src={italyFlag}
          alt="italiano"
          onClick={() => this.context.onLanguageChange("it")}
        />
        <img
          src={spainFlag}
          alt="espaÃ±ol"
          onClick={() => this.context.onLanguageChange("es")}
        />
      </div>
    );
  }
}

export default LanguageSelector;
