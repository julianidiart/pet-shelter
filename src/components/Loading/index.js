import React from "react";
import loader from "../../assets/images/loader.gif";

const Loading = () => (
  <div className="loader">
    <img className="loader__image" src={loader} alt="loader" />
  </div>
);

export default Loading;
