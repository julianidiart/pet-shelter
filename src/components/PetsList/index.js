import React from "react";
import PetsListItem from "./PetsListItem";
import { connect } from "react-redux";

export const PetsList = ({ pets }) => (
  <div>
    {pets.length === 0 ? (
      <p>Add a pet to start!</p>
    ) : (
      pets &&
      pets.map(pet => {
        return <PetsListItem key={pet.id} {...pet} />;
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    pets: state.pets
  };
};

export default connect(mapStateToProps)(PetsList);
