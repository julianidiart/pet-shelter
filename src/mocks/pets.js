export default [
  { id: "1", name: "Marla" },
  { id: "2", name: "Bimba" },
  { id: "3", name: "Aimee" },
  { id: "4", name: "Maia" },
  { id: "5", name: "Emma" },
  { id: "6", name: "Tommy" },
  { id: "7", name: "Bello Bruno" },
  { id: "8", name: "Bubu" },
  { id: "9", name: "Blaky" }
];
