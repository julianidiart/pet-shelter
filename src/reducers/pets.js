const petsReducerDefaultState = [];

export default (state = petsReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_PET":
      return [...state, action.pet];
    case "REMOVE_PET":
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_PET":
      return state.map(pet => {
        if (pet.id === action.id) {
          return {
            ...pet,
            ...action.pet
          };
        } else {
          return pet;
        }
      });
    case "SET_PETS":
      return action.pets;
    default:
      return state;
  }
};
