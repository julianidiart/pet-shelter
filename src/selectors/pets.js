import moment from "moment";

export default (pets, { text, sortBy, startDate, endDate, filter }) => {
  return pets
    .filter(pet => {
      const birthdateMoment = moment(pet.birthdate);
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(birthdateMoment, "day")
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(birthdateMoment, "day")
        : true;
      const textMatch =
        pet.name.toLowerCase().includes(text.toLowerCase()) ||
        pet.place.toLowerCase().includes(text.toLowerCase()) ||
        pet.breed.toLowerCase().includes(text.toLowerCase()) ||
        (pet.comments &&
          pet.comments.toLowerCase().includes(text.toLowerCase())) ||
        (pet.color
          ? pet.color.toLowerCase().includes(text.toLowerCase())
          : false) ||
        pet.chip
          .toString()
          .toLowerCase()
          .includes(text.toLowerCase());
      let filterMatch = true;
      const adoptedOrPassedAway = pet.adopted || pet.passedAway;
      switch (filter) {
        case "sterilized":
          filterMatch = !adoptedOrPassedAway && pet.sterilized === true;
          break;
        case "non-sterilized":
          filterMatch = !adoptedOrPassedAway && pet.sterilized === false;
          break;
        case "chipped":
          filterMatch = !adoptedOrPassedAway && pet.chip !== "";
          break;
        case "non-chipped":
          filterMatch = !adoptedOrPassedAway && pet.chip === "";
          break;
        case "adopted":
          filterMatch = pet.adopted === true;
          break;
        case "passed-away":
          filterMatch = pet.passedAway === true;
          break;
        default:
          filterMatch = !adoptedOrPassedAway;
      }

      return startDateMatch && endDateMatch && textMatch && filterMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.birthdate < b.birthdate ? 1 : -1;
      }
      return a.name < b.name ? 1 : -1;
    });
};
