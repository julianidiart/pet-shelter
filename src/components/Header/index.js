import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../../actions/auth";
import LanguageSelector from "../LanguageSelector";
import MultiLanguageText from "../MultiLanguageText";

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/pets">
          <h1>Pet Shelter</h1>
        </Link>
        <div>
          <Link className="button button--link" to="/pets">
            <MultiLanguageText en="Pets" it="Animali" es="Mascotas" />
          </Link>
          <Link className="button button--link" to="/volunteers">
            <MultiLanguageText
              en="Volunteers"
              it="Volontari"
              es="Voluntarios"
            />
          </Link>
          <button className="button button--link" onClick={startLogout}>
            <MultiLanguageText en="Logout" it="Esci" es="Salir" />
          </button>
        </div>
        <LanguageSelector />
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
