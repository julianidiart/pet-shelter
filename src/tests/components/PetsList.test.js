import React from "react";
import { shallow } from "enzyme";
import { PetsList } from "../../components/PetsList";
import pets from "../fixtures/pets";

test("should render PetsList with pets", () => {
  const wrapper = shallow(<PetsList pets={pets} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render PetsList with empty message", () => {
  const wrapper = shallow(<PetsList pets={[]} />);
  expect(wrapper).toMatchSnapshot();
});
