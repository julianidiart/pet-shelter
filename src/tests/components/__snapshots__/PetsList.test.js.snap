// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PetsList with empty message 1`] = `
<div>
  <p>
    Add a pet to start!
  </p>
</div>
`;

exports[`should render PetsList with pets 1`] = `
<div>
  <_default
    birthdate={-63158400000}
    chip="123ABC123"
    id="1"
    images={
      Array [
        "https://www.pet-art.net/images/pastel/labrador-retriever-jenson1.jpg",
      ]
    }
    key="1"
    name="Marla"
    place="Barchetta"
    sex="F"
  />
  <_default
    birthdate={-47437200000}
    chip="123ABC123"
    id="2"
    key="2"
    name="Bimba"
    place="Barchetta"
    sex="F"
  />
  <_default
    birthdate={-15901200000}
    chip="123ABC123"
    id="3"
    key="3"
    name="Aimee"
    place="Barchetta"
    sex="F"
  />
  <_default
    birthdate={-110595600000}
    id="4"
    key="4"
    name="Maia"
    place="Campagna"
    sex="F"
  />
  <_default
    birthdate={-347155200000}
    id="5"
    key="5"
    name="Emma"
    place="Barchetta"
    sex="F"
  />
  <_default
    birthdate={-252460800000}
    chip="123ABC123"
    id="6"
    key="6"
    name="Tommy"
    place="Macelo"
    sex="M"
  />
  <_default
    birthdate={-283996800000}
    id="7"
    key="7"
    name="Bello Bruno"
    place="Sasi"
    sex="M"
  />
  <_default
    birthdate={-126230400000}
    chip="123ABC123"
    id="8"
    key="8"
    name="Bubu"
    place="Sasi"
    sex="M"
  />
  <_default
    birthdate={-157766400000}
    id="9"
    key="9"
    name="Blaky"
    place="Macelo"
    sex="M"
  />
</div>
`;
